vignettes/develop.Rmd

```diff
@@ -0,0 +1,131 @@
+---
+title: "Developing rirods"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Developing rirods}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Contributing to rirods development
+
+To contribute to rirods fork the source files from: 
+https://github.com/irods/irods_client_library_rirods, and make a Pull Request. 
+
+## Prerequisites R
+
+The following R packages are required for development: `jsonlite`, `readr`, 
+`testthat` (>= 3.0.0) and `withr`, and `httr2` and `httptest2`. For convenience,
+one can also use `renv` () and `renv::restore()` to install all 
+dependencies. The `usethis` and the `devtools` packages are further 
+suggestions to streamline development. See https://devtools.r-lib.org/ and 
+https://usethis.r-lib.org/ for common development tasks and setups covered by 
+these packages.
+
+Development of `rirods` relies on the `httptest2` which enables testing of the 
+logic on the R sides of the API without requiring access to the remote service.
+It does this by recording the API response as mock files. These mock files are 
+stored in several directories under `.tests/testthat/`. To enable testing
+against the real API one needs to remove these mock files. The helper 
+`remove_mock_files()` simplifies this task. This function is not exported and 
+needs to be sourced (e.g., using `devtools::load_all()`) and subsequently 
+called.
+
+## Prerequisites iRODS
+
+The unit test included in the source files can either be run against a local or 
+remote iRODS server. The iRODS server should include the iRODS C++ REST API of 
```

Comment:
"remote iRODS server. The iRODS server should include the iRODS C++ REST API of "
->
"remote iRODS C++ REST API service, version "